### DISCUSSION TITLE
Issues using mpirun

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#16677] [Q&A General] [answered]
Hi, 

I am having issues with using mpirun on Ubuntu
The command I am using is this following
`mpirun -n 16 <my-application-executable> -i <my input file>`

This gives the following chain of errors 
[errorex.txt](https://github.com/idaholab/moose/files/5809415/errorex.txt)

When running `mpirun -n 2 echo 'hello world'` I get the intended output of two echos of `hello world`.

The `gethostbyname` error made me believe the issue was with my /etc/hosts/ file. However, despite modifications, I have the same error. 

The file currently looks like this
[host.txt](https://github.com/idaholab/moose/files/5809433/host.txt)

The `det12` is my username. I have also tried using the hostname for my computer. 

I have tried replacing the det12 with `det12.local`, `<my-computer-hostname>`, and `<my-computer-hostname>.local`

Nothing seems to have worked. Everything has been done in the moose environment.
Any insight into this would be much appreciated.

**Comment by loganharbour** [comment, marked as answer]
You should add an entry in your hosts file for the host that appears in the failed MPI run:
```
MPID_nem_tcp_init(373).............: gethostbyname failed, pse-jdolbow-hp8rq53 (errno 0)
```

In this case, you should add
```
127.0.0.1 pse-jdolbow-hp8rq53
```

**Reply by TheGreatCid** [reply]
I've tried that as well, with this host file. The same error message displays after using mpirun
[host.txt](https://github.com/idaholab/moose/files/5810029/host.txt)

**Reply by milljm** [reply]
Searching this and specifically Linux, yielded this result:

  https://ubuntuforums.org/showthread.php?t=1921181 (last response). The important bits:

```
127.0.0.1	localhost
127.0.0.1	zip.chem.rochester.edu
127.0.1.1	zip
127.0.1.1	128.151.187.109
128.151.187.109 zip
128.151.187.109 zip.chem.rochester.edu
  
# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
ff02::3 ip6-allhosts
```

Supposedly, you should list all possible addresses (not just localhost of 127.0.0.1). Why you might have to do this is a mystery. I've never seen this issue occur on a Linux machine myself.

**Reply by TheGreatCid** [reply]
That is really odd. 
Even odder was that the issue was fixed when I connected my computer to the internet. Really doesn't make sense considering that everything is local, but I'm not complaining. 
Still thank you for the additional advice.
